core/interop/pinvoke/marshalling/src/MarshallingSampleNative/MarshallingSampleNative.cpp

```diff
@@ -0,0 +1,356 @@
+#include "common.h"
+#include <string>
+#include <string.h>
+#include <stdlib.h>
+
+// string functions
+
+extern "C" DLL_EXPORT int STDMETHODCALLTYPE CountUtf8Characters(char* value)
+{
+    // value is ANSI (on Windows)/ UTF8 (on Linux/Mac) null-terminated string
+    // and can be null.
+    if (value == nullptr)
+    {
+        return -1;
+    }
+    else
+    {
+        return strlen(value);
```

Comment:
`CountUtf8Characters()` is a questionable name for this API. `strlen` only counts `char` units. For UTF8, the correct word would be codepoints, but that isn't what `strlen` does. I would instead change to ByteCount or something that doesn't include the expected encoding - i.e. UTF8.